mqclient/queue.py

```diff
@@ -302,6 +393,13 @@ def __iter__(self) -> "MessageGeneratorContext":
             raise RuntimeError(self.RUNTIME_ERROR_CONTEXT_STRING)
         return self
 
+    @wtt.spanned(
+        # TODO - add link/parent as `self._span`
+        these=["self.i"],
+        kind=wtt.SpanKind.CONSUMER,
+        # carrier="properties.headers",  # TODO: figure how to get carrier
+        carrier_relation=wtt.CarrierRelation.LINK,
+    )
```

Comment:
Since you asked about event vs span, my opinion on this is that a span would be useful but not critical.  I do like the duration, but I guess we could get at that in other ways as well (especially if the calling code is instrumented).

It is a little unclear what the `self.i` is actually doing, besides being a counter.  That seems less than ideal, and may point to an event being better.

Reply:
Yup, that's essentially what it came down to, durations. We'll have to see in production if this can be a bottleneck. I'm not too sure how the OTEL exporter handles batches of spans.

Re: `self.i`, this is just for the span. It'll be useful to see if we're handling dozens vs thousands of messages/iterations. Though this can also be decerned by looking at the number of sibling spans, along with timestamps.

Reply:
Considering that counting is one thing monitoring frameworks are really good at themselves, it seems not great to do it ourselves, unless we need a unique id for the spans.

Reply:
Great point, it's nixed!